cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/index.tsx

```diff
@@ -50,7 +53,11 @@ const DeployedPipeline: React.FC = () => {
   // on unmount
   React.useEffect(() => {
     return () => {
-      reset();
+      // (TODO): This is right now problematic.
+      // React unmounts component everytime when incoming props change
+      // Because this is FC there is no way to know if the parent component
+      // changed the props or really unmounted.
+      // reset();
```

Comment:
so where is the reset happening now?

Reply:
It is not. Thats why added the comment. We right now don't reset the store.

Reply:
that doesn't cause issue in navigating back and forth while staying in React components?

Reply:
Search and sorting will exist but the pipelines won't.  I will have to reset but not erase the pipelines and filtered ones. Still trying it out.

Reply:
Fixed. Can you give it another review.